base/compiler/ssair/slot2ssa.jl

```diff
@@ -675,11 +633,18 @@ function construct_ssa!(ci::CodeInfo, ir::IRCode, domtree::DomTree,
                 # an upsilon node in the corresponding enter block
                 node = PhiCNode(Any[])
                 insertpoint = first_insert_for_bb(code, cfg, li)
+                @assert sv.bb_vartables[li] !== nothing
+                vt = sv.bb_vartables[li][idx]
+                typ = widenslotwrapper(ignorelimited(vt.typ))
```

Comment:
We may want to `widenslotwrapper` within `type_annotate!` and force `sv.bb_vartables` to be valid for `optimizer_lattice` always?

Reply:
Yeah, great idea - Let's just widen the types at the inference boundary like we do elsewhere